.travis.yml

```diff
@@ -72,12 +78,60 @@ before_install:
     - tools/check_sdist.py
 
 install:
-    - section build
     - python setup.py develop
-    - section_end build
+    # Matplotlib settings - do not show figures during doc examples
+    - |
+      if [[ "${MINIMUM_REQUIREMENTS}" == "1" || "${TRAVIS_OS_NAME}" == "osx" ]]; then
+        export MPL_DIR=${HOME}/.matplotlib
+      else
+        export MPL_DIR=${HOME}/.config/matplotlib
+      fi
+    - mkdir -p ${MPL_DIR}
+    - touch ${MPL_DIR}/matplotlibrc
+    # Install most of the optional packages
+    - |
+      if [[ "${OPTIONAL_DEPS}" == "1" ]]; then
+        pip install --retries 3 -q -r ./requirements/optional.txt $WHEELHOUSE
+      fi
+    - |
+      if [[ "${WITH_PYAMG}" == "1" ]]; then
+        pip install --retries 3 -q pyamg
+      fi
+    - source tools/travis/install_qt.sh
 
-script: tools/travis/script.sh
+script:
+    - export PY=${TRAVIS_PYTHON_VERSION}
+    - flake8 --exit-zero --exclude=test_*,six.py skimage doc/examples viewer_examples
+    # run tests again with optional dependencies to get more coverage
+    - |
+      if [[ "$OPTIONAL_DEPS" == "1" ]]; then
+        export TEST_ARGS="${TEST_ARGS} --cov=skimage"
+      fi
+    # Show what's installed
+    - pip list
+    - pytest ${TEST_ARGS} skimage
+    # Run example applications
+    - echo Build or run examples
+    - |
+      if [[ "${BUILD_DOCS}" == "1" ]]; then
+        # requirements/docs.txt fails on Travis OSX
+        pip install --retries 3 -q -r ./requirements/docs.txt
+        export SPHINXCACHE=${HOME}/.cache/sphinx; make html
+      elif [[ "${TEST_EXAMPLES}" != "0" ]]; then
+        pip install --retries 3 -q -r ./requirements/docs.txt
+        cp $MPL_DIR/matplotlibrc $MPL_DIR/matplotlibrc_backup
+        echo 'backend : Template' > $MPL_DIR/matplotlibrc
+        for f in doc/examples/*/*.py; do
+            python "${f}"
+            if [ $? -ne 0 ]; then
+                exit 1
+            fi
+        done
+        mv $MPL_DIR/matplotlibrc_backup $MPL_DIR/matplotlibrc
+      fi
```

Comment:
Why did you change back from a script to travis.yml lines?

Reply:
I was trying to get little helpful timings to show up:

![image](https://user-images.githubusercontent.com/90008/40209214-dbefef04-59f2-11e8-9076-3e3859ad2345.png)
I"m not sure if it is a bug with travis or not. but the "script" didn't have those cool folding sections.

Reply:
They do some some kind of timing per command, but it isn't nearly as helpful.

I'll change it back to the script.

https://travis-ci.org/scikit-image/scikit-image/jobs/379492893#L4072

Reply:
But it doesn't work really. The script one doesn't fold as nicely. I don't know why.

Reply:
We have this in a script because editing it is much less painful than updating a YAML list of commands.

Reply:
@stefanv yes editing the YAML was painful. I changed it back.